src/ADService.js

```diff
@@ -55,16 +55,33 @@ class ADService {
     new Date().getTime() < tokenResult.expiresOn * 1000;
 
   getAccessTokenAsync = async () => {
-    if (!this._isTokenValid(this.tokenResult)) {
-      const result = await this.fetchAndSetTokenAsync(
-        this.tokenResult.refreshToken,
-        this.loginPolicy,
-        true
-      );
-
-      if (!result.isValid) {
-        return result;
+    try {
+      // Additional is valid to store result from second request
+      let isValid = false;
+      if (!this._isTokenValid(this.tokenResult)) {
+        const result = await this.fetchAndSetTokenAsync(
+          this.tokenResult.refreshToken,
+          this.loginPolicy,
+          true
+        );
+
+        /*
+         * If wrong endpoint try with reset password endpoint
+         */
+        if (result.data && result.data.match("AADB2C90088")) {
+          const result2 = await this.fetchAndSetTokenAsync(
+            this.tokenResult.refreshToken,
+            this.passwordResetPolicy,
+            true
+          );
+          isValid = result2.isValid;
+        }
+        if (!result.isValid && !isValid) {
+          return result;
```

Comment:
Something seems off here.

1. Shouldn't this function return `result2` if `(result.data && result.data.match("AADB2C90088"))` and `result2.isValid`?
2. Should this function return `result` if `result.isValid`? This function returns `result` only if both requests are invalid?

Reply:
1. No, this is a kind of weird thing i found out after our discussion at my place last week. So, if the `fetchAndSetTokenAsync` method succeeds the result does not contain the access token in the data field (that's why i had `true, undefined` in the result). 
2. The first clause is to check if there is an error: if there is an error return the error. In line 90 the token is collected and returned if the earlier checks pass

Reply:
1. if there is something fishy, you should describe the reasoning here. Since it doesn't make that much sense by just reading the code.
2. oki